ext-meta-generator/src/main/java/tech/ydb/yoj/generator/FieldGeneratorAnnotationProcessor.java

```diff
@@ -0,0 +1,181 @@
+ *          public static class IdField {
+ *              public static final String VALUE = "idField.value"; // Mind that the value has "idField." prefix
+ *
+ *              // The Annotation Processor works recursively
+ *              // Also it's the example of several fields with the same type
+ *              static class DeepField1 {
+ *                  public static final String FIELD = "idField.deepField1.field";
+ *              }
+ *              static class DeepField2 {
+ *                  // Pay attention that ".deepField2." is used here
+ *                  public static final String FIELD = "idField.deepField2.field";
+ *              }
+ *          }
+ *      }
+ *  }
+ *  <ul>
+ *      <b>Additional info:</b>
+ *      <li>Support Records</li>
+ *  </ul>
+ * <ul>
+ *      <b>Known issues (should be fixed in future):</b>
+ *     <li>If there is Id-class with a single field id, it will generate "id.id" constant instead of "id" which is
+ *     assumed in our ORM</li>
+ *     <li>if entity doesn't have @Table it won't be processed even if it's implements Entity interface</li>
+ *     <li>We assume that annotation @Table is used on top-level class</li>
+ *     <li>The AP will break in case of two nested classes which refer each other (i.e. circular dependency) </li>
+ *     <li>Will generate nested classes even if @Column(flatten=true) </li>
+ *     <li>Will generate nested classes even if they are in fact empty </li>
+ *     <li>No logs are written</li>
+ *     <li>There is a rare situation when generated code won't compile. The following source class
+ *     <pre>{@code
+ *          class Name{
+ *              Class1 nameClash;
+ *              public static class Class1 {
+ *                  Class2 nameClash;
+ *                  class Class2{
+ *                      String nameClash;
+ *                  }
+ *              }
+ *          }
+ *     }
+ *     will produce
+ *     <pre>{@code
+ *         public class NameFields {
+ *             public class NameClash {
+ *                 public class NameClash {
+ *                     public static final String NAME_CLASH = "nameClash.nameClash.nameClash";
+ *                 }
+ *             }
+ *         }
+ *     }
+ *     which won't compile due to 2 NameClash classes.
+ *     </li>
+ * </ul>
+ *
+ * @author pavel-lazarev
+ */
+
+// Test When a class A has B, B has C. And A uses B.C as field Type
+@SupportedAnnotationTypes({
+        "tech.ydb.yoj.databind.schema.Table",
+})
+@SupportedSourceVersion(SourceVersion.RELEASE_17)
+public class FieldGeneratorAnnotationProcessor extends AbstractProcessor {
+
+    private static final Logger log = LoggerFactory.getLogger(FieldGeneratorAnnotationProcessor.class);
+
+    private static final String TARGET_PACKAGE = "generated";
+    private static final String TARGET_CLASS_NAME_SUFFIX = "Fields";
+
+    @Override
+    public boolean process(Set<? extends TypeElement> set, RoundEnvironment roundEnvironment) {
```

Comment:
question: Will it be run on every compilation? It could take a long time on large project. Could we calculate hash of file and recreate a files only of hash is changed?